nvidia-ubuntu/Dockerfile

```diff
@@ -5,30 +5,27 @@ LABEL maintainer="David Beckingsale <david@llnl.gov>,vsoch"
 
 ENV cuda_version=$cuda_version
 ENV CUDA_TOOLKIT_ROOT_DIR=/usr/local/cuda-${cuda_version}
+ENV spack_commit=develop
 
 ENV DEBIAN_FRONTEND=noninteractive
 ENV TZ=America/Los_Angeles
 
-RUN apt-get -qq update && \
-    apt-get -qq install -y --no-install-recommends \
-      wget \
-      python3-dev \
-      python3-pip \
-      build-essential \
-      sudo \
-      git \
-      vim \
-      dh-autoreconf \
-      ninja-build \
-      ca-certificates
+WORKDIR /opt
+COPY ./scripts /opt/scripts
+COPY ./ubuntu/scripts/* /opt/scripts
+RUN ./scripts/apt-install-defaults-plus-args
+ENV CMAKE=3.20.4
+RUN ./scripts/install-cmake-binary
 
 # Install spack
-RUN git clone --depth 1 https://github.com/spack/spack /opt/spack
+RUN ./scripts/set-up-spack
 ENV PATH=/opt/spack/bin:$PATH
+# Tell spack to use this one without arguments
+ENV SPACK_ENV=/opt/env
 
-# Install cmake 3.10
-RUN wget -q --no-check-certificate https://cmake.org/files/v3.10/cmake-3.10.1-Linux-x86_64.tar.gz && \
-    tar -xzf cmake-3.10.1-Linux-x86_64.tar.gz && \
-    cp -fR cmake-3.10.1-Linux-x86_64/* /usr && \
-    rm -rf cmake-3.10.1-Linux-x86_64 && \
-    rm cmake-3.10.1-Linux-x86_64.tar.gz
+# Add appropriate paths pointing to the view
```

Comment:
Ok, I think that should be fine.  I normally order by when something would be valid, but since these are search paths it shouldn't matter, they just wont have anything in them yet.